apps/cowswap-frontend/src/modules/twap/hooks/useCreateTwapOrder.tsx

```diff
@@ -97,6 +97,9 @@ export function useCreateTwapOrder() {
         const fallbackSetupTxs = fallbackHandlerIsNotSet
           ? await extensibleFallbackSetupTxs(extensibleFallbackContext)
           : []
+
+        // upload the app data here, as application might need it to decode the order info before it is being signed
+        uploadAppData({ chainId, orderId, appData: appDataInfo })
         const createOrderTxs = createTwapOrderTxs(twapOrder, paramsStruct, twapOrderCreationContext)
         const { safeTxHash } = await safeAppsSdk.txs.send({ txs: [...fallbackSetupTxs, ...createOrderTxs] })
```

Comment:
nit: we could consider deleting it if this `txs.send(...)` fails, right? (i.e. user rejects the transaction). Maybe this helps with the pollution?